File: src/main/java/dev/sky_lock/pocketlifevehicle/command/SearchCommand.java
package dev.sky_lock.pocketlifevehicle.command;

import dev.sky_lock.pocketlifevehicle.PLVehicle;
import dev.sky_lock.pocketlifevehicle.Permission;
import dev.sky_lock.pocketlifevehicle.vehicle.Car;
import dev.sky_lock.pocketlifevehicle.vehicle.CarEntities;
import dev.sky_lock.pocketlifevehicle.util.Profiles;
import org.bukkit.ChatColor;
import org.bukkit.Location;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import java.util.UUID;

/**
 * @author sky_lock
 */

public class SearchCommand implements ICommand {

    @Override
    public void execute(CommandSender sender, Command cmd, String[] args) {
        Player player = (Player) sender;
        if (!Permission.ADMIN_COMMAND.obtained(player) || args.length < 2) {
            Location location = getCarLocation(player.getUniqueId());
            if (location == null) {
                player.sendMessage(PLVehicle.PREFIX + ChatColor.RED + "車を所有していません");
            } else {
                sendLocation(player, location);
            }
            return;
        }

        String name = args[1];
        UUID targetUUID = Profiles.fetchUUID(name);
        Location location = getCarLocation(targetUUID);
        if (location == null) {
            player.sendMessage(PLVehicle.PREFIX + ChatColor.RED + "指定したプレイヤーは車を所有していません");
        } else {
            sendLocation(player, location);
        }
    }

    private Location getCarLocation(UUID target) {
        Car car = CarEntities.of(target);
        if (car == null) {
            return null;
        }
        return car.getLocation();
    }

    private void sendLocation(Player player, Location loc) {
        player.sendMessage(PLVehicle.PREFIX + ChatColor.GREEN + "-------------------------------------------");
        player.sendMessage(PLVehicle.PREFIX + ChatColor.GREEN + "World : " + loc.getWorld().getName());
        player.sendMessage(PLVehicle.PREFIX + ChatColor.GREEN + "X : " + loc.getBlockX());
        player.sendMessage(PLVehicle.PREFIX + ChatColor.GREEN + "Y : " + loc.getBlockY());
        player.sendMessage(PLVehicle.PREFIX + ChatColor.GREEN + "Z : " + loc.getBlockZ());
        player.sendMessage(PLVehicle.PREFIX + ChatColor.GREEN + "-------------------------------------------");
    }
}

File: src/main/kotlin/dev/sky_lock/pocketlifevehicle/command/SearchCommand.kt
package dev.sky_lock.pocketlifevehicle.command

import dev.sky_lock.pocketlifevehicle.Permission
import dev.sky_lock.pocketlifevehicle.extension.chat.sendVehiclePrefixedErrorMessage
import dev.sky_lock.pocketlifevehicle.extension.chat.sendVehiclePrefixedSuccessMessage
import dev.sky_lock.pocketlifevehicle.vehicle.VehicleManager
import org.bukkit.Bukkit
import org.bukkit.Location
import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

/**
 * @author sky_lock
 */
class SearchCommand : ICommand {
    override fun execute(sender: CommandSender, cmd: Command, args: Array<String>) {
        val player = sender as Player
        if (!Permission.ADMIN_COMMAND.obtained(player) || args.size < 2) {
            val location = VehicleManager.getLocation(player.uniqueId)
            if (location == null) {
                player.sendVehiclePrefixedErrorMessage( "乗り物の現在地を取得できませんでした")
                return
            }
            if (location.world == player.location.world) {
                player.sendVehiclePrefixedSuccessMessage( "乗り物は現在(" + getLocationString(location) + ")にあります")
                return
            }
            player.sendVehiclePrefixedErrorMessage( "別ワールドにある乗り物の現在地は取得できません")
            return
        }
        val name = args[1]
        val targetUUID = Bukkit.getPlayerUniqueId(name)
        if (targetUUID == null) {
            player.sendVehiclePrefixedErrorMessage( "プレイヤーが見つかりませんでした")
            return
        }
        val location = VehicleManager.getLocation(targetUUID)
        if (location == null) {
            player.sendVehiclePrefixedErrorMessage( "プレイヤーの乗り物の現在地を取得できませんでした")
            return
        }
        player.sendVehiclePrefixedSuccessMessage( "(world=" + location.world.name + ", " + getLocationString(location) + ")")
    }

    private fun getLocationString(loc: Location): String {
        return "x=" + loc.blockX.toString() + ", y=" + loc.blockY.toString() + ", z=" + loc.blockZ.toString()
    }
}